x/ecocredit/core/msg_credit_type.go

```diff
@@ -33,8 +33,7 @@ func (m CreditType) Validate() error {
 // uppercase letters
 func validateCreditTypeAbbreviation(abbr string) error {
 	reAbbr := regexp.MustCompile(`^[A-Z]{1,3}$`)
-	matches := reAbbr.FindStringSubmatch(abbr)
-	if matches == nil {
+	if !reAbbr.Match([]byte(abbr)) {
```

Comment:
switched to the single match, should be a bit faster

Reply:
We should remove this private function and use core utils `ValidateCreditTypeAbbreviation`.